pkg/batcher/controllers/default.go

```diff
@@ -0,0 +1,307 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/go-logr/logr"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+	"errors"
+	"context"
+	"encoding/json"
+	"github.com/astaxie/beego"
+	"github.com/satori/go.uuid"
+)
+
+const (
+	SleepTime    = time.Microsecond * 100
+	MaxBatchsize = 32
+	MaxLatency   = 1.0
+)
+
+var (
+	channelIn = make(chan Input)
+	batcherInfo BatcherInfo
+	batcherHandler *BatcherHandler = nil
+	mutex sync.Mutex
+)
+
+type MainController struct {
+	beego.Controller
+}
+
+type Request struct {
+	Instances []interface{} `json:"instances"`
+}
+
+type Input struct {
+	ContextInput *context.Context `json:"contextInput"`
+	Instances *[]interface{} `json:"instances"`
+	ChannelOut *chan Response `json:"channelOut"`
+}
+
+type InputInfo struct {
+	ChannelOut *chan Response `json:"channelOut"`
+	Index []int `json:"index"`
+}
+
+type Response struct {
+	BatchID string `json:"batchId"`
+	Predictions []interface{} `json:"predictions"`
+}
+
+type ResponseError struct {
+	Message string `json:"message"`
+}
+
+type Predictions struct {
+	Predictions []interface{} `json:"predictions"`
+}
+
+type BatcherInfo struct {
+	MaxBatchsize int `json:"maxBatchsize"`
+	MaxLatency float64 `json:"maxLatency"`
+	BatchID string `json:"batchId"`
+	Instances []interface{} `json:"instances"`
+	Predictions Predictions `json:"predictions"`
+	Info map[*context.Context] InputInfo `json:"info"`
+	Start time.Time `json:"start"`
+	Now time.Time `json:"now"`
+	CurrentInputLen int `json:"currentInputLen"`
+}
+
+type BatcherHandler struct {
+	Log              logr.Logger
+	Port             string
+	SvcHost          string
+	SvcPort          string
+	MaxBatchsize     int
+	MaxLatency       float64
+	Path             string
+	ContentTpye      string
+}
+
+func New(log logr.Logger, port string, svcHost string, svcPort string,
+	maxBatchsize int, maxLatency float64) {
+	batcherHandler = &BatcherHandler{
+		Log:              log,
+		Port:             port,
+		SvcHost:          svcHost,
+		SvcPort:          svcPort,
+		MaxBatchsize:     maxBatchsize,
+		MaxLatency:       maxLatency,
+		Path:             "",
+		ContentTpye:      "",
+	}
+}
+
+func GetNowTime() time.Time {
+	return time.Now().UTC()
+}
+
+func GenerateUUID() string {
+	if uuidStr, err := uuid.NewV4(); err == nil {
+		return uuidStr.String()
+	}
+	return ""
+}
+
+func CallService() {
+	var errStr string
+	url := fmt.Sprintf("http://%s:%s%s", batcherHandler.SvcHost, batcherHandler.SvcPort, batcherHandler.Path)
+	jsonStr, _ := json.Marshal(Request{
```

Comment:
we may need to check header here if the original request is sent in as json

Reply:
I will modify it.